src/xo-app/new-vm/index.js

```diff
@@ -239,6 +239,20 @@ export default class NewVm extends BaseComponent {
     resolveResourceSet
   )
 
+  _getCoresPerSocketPossibilities = () => {
+    // According to : https://www.citrix.com/blogs/2014/03/11/citrix-xenserver-setting-more-than-one-vcpu-per-vm-to-improve-application-performance-and-server-consolidation-e-g-for-cad3-d-graphical-applications/
+    const maxVCPUs = 16
+    const maxCoresPerSocket = this.props.pool.cpus.cores
+    const vCPUs = this.state.state.CPUs
+
+    const options = []
+    for (let coresPerSocket = 1; coresPerSocket <= maxCoresPerSocket; coresPerSocket++) {
+      if (vCPUs !== '' && vCPUs % coresPerSocket === 0 && vCPUs / coresPerSocket <= maxVCPUs) options.push(coresPerSocket)
+    }
+
+    return options
+  }
```

Comment:
Memoize this selector (`createSelector`), no need to keep recomputing if its input does not change.